src/transformers/models/colpali/processing_colpali.py

```diff
@@ -0,0 +1,152 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Union
+
+import torch
+import torch.utils.checkpoint
+from PIL import Image
+
+from ...feature_extraction_utils import BatchFeature
+from ..paligemma import (
+    PaliGemmaProcessor,
+)
+
+
+class ColPaliProcessor(PaliGemmaProcessor):
+    r"""
+    Processor for ColPali.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mock_image = Image.new("RGB", (16, 16), color="black")
+
+    @staticmethod
+    def get_torch_device(device: str = "auto") -> str:
+        """
+        Returns the device (string) to be used by PyTorch.
+
+        `device` arg defaults to "auto" which will use:
+        - "cuda:0" if available
+        - else "mps" if available
+        - else "cpu".
+        """
+
+        if device == "auto":
+            if torch.cuda.is_available():
+                device = "cuda:0"
+            elif torch.backends.mps.is_available():  # for Apple Silicon
+                device = "mps"
+            else:
+                device = "cpu"
+            logger.info(f"Using device: {device}")
+
+        return device
+
+    def process_images(
+        self,
+        images: List[Image.Image],
+    ) -> BatchFeature:
+        """
+        Process images for ColPali.
+        """
+        texts_doc = ["Describe the image."] * len(images)
+        images = [image.convert("RGB") for image in images]
+
+        batch_doc = self(
+            text=texts_doc,
+            images=images,
+            return_tensors="pt",
+            padding="longest",
+        )
+        return batch_doc
+
+    def process_queries(
+        self,
+        queries: List[str],
+        max_length: int = 50,
+        suffix: Optional[str] = None,
+    ) -> BatchFeature:
+        """
+        Process queries for ColPali.
+        """
+        if suffix is None:
+            suffix = "<pad>" * 10
+        texts_query: List[str] = []
+
+        for query in queries:
+            query = f"Question: {query}"
+            query += suffix  # add suffix (pad tokens)
+            texts_query.append(query)
+
+        batch_query = self(
+            images=[self.mock_image] * len(texts_query),
+            text=texts_query,
+            return_tensors="pt",
+            padding="longest",
+            max_length=max_length + self.image_seq_length,
+        )
+
+        del batch_query["pixel_values"]
+
+        batch_query["input_ids"] = batch_query["input_ids"][..., self.image_seq_length :]
+        batch_query["attention_mask"] = batch_query["attention_mask"][..., self.image_seq_length :]
+
+        return batch_query
```

Comment:
Could you try overriding the `__call__` function of `PaliGemma` processor instead of creating these new functions?

Reply:
I'll hypothesize what you might have in mind: a `__call__` method with both `images` and `queries` as input args. Am I wrong?

Reply:
Yep something like that! We are in the process of uniformizing processors for multimodal models, so that would include ColPali. you can take a look at the processor of [PaliGemma ](https://github.com/huggingface/transformers/blob/b1c914e463b4136e598f1d8c5248e87c46bd19bf/src/transformers/models/paligemma/processing_paligemma.py) to have an idea of how the __call__ function and the handling of kwargs should look like!